storage-proofs-porep/src/lib.rs

```diff
@@ -1,4 +1,5 @@
 #![deny(clippy::all, clippy::perf, clippy::correctness, rust_2018_idioms)]
+#![allow(clippy::mut_from_ref)]
```

Comment:
It seems that if `UnsafeSlice::data` is renamed to `_data`, that this isn't needed anymore.

Reply:
No, it relates to a different clippy's caprice:

```
error: mutable borrow from immutable input(s)
  --> storage-proofs-porep/src/stacked/vanilla/utils.rs:26:42
   |
26 |     pub unsafe fn as_mut_slice(&self) -> &mut [T] {
   |                                          ^^^^^^^^
   |
note: the lint level is defined here
  --> storage-proofs-porep/src/lib.rs:1:36
   |
1  | #![deny(clippy::all, clippy::perf, clippy::correctness, rust_2018_idioms)]
   |                                    ^^^^^^^^^^^^^^^^^^^
   = note: `#[deny(clippy::mut_from_ref)]` implied by `#[deny(clippy::correctness)]`
note: immutable borrow here
  --> storage-proofs-porep/src/stacked/vanilla/utils.rs:26:32
   |
26 |     pub unsafe fn as_mut_slice(&self) -> &mut [T] {
   |                                ^^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#mut_from_ref

error: mutable borrow from immutable input(s)
  --> storage-proofs-porep/src/stacked/vanilla/utils.rs:41:51
   |
41 |     pub unsafe fn get_mut(&self, index: usize) -> &mut T {
   |                                                   ^^^^^^
   |
note: immutable borrow here
  --> storage-proofs-porep/src/stacked/vanilla/utils.rs:41:27
   |
41 |     pub unsafe fn get_mut(&self, index: usize) -> &mut T {
   |                           ^^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#mut_from_ref
```
(from [this](https://app.circleci.com/pipelines/github/filecoin-project/rust-fil-proofs/3991/workflows/5c13e933-8b1f-4965-9e2a-0640c51b2975/jobs/90034) build)

Reply:
It works for me locally without it.

Reply:
Yes, indeed, on my development machine it works too for some reasons. But things are a bit different on CI's container. I had to debug those issues just according to output from CI

Reply:
Yep. Agree